src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -417,6 +418,13 @@ public void run() {
             long timeout = verifyTimeout * 1000;
             long endTime = System.currentTimeMillis() + timeout;
 
+            // Check for port already in use error
+            String portError = serverTask.waitForStringInLog(PORT_IN_USE_MESSAGE_REGEXP, timeout, messagesLogFile);
+            if (portError != null) {
+            	stopServer();
+            	throw new PluginExecutionException("Configured port may already be in use.");
```

Comment:
Does it reach this message after calling stopServer() above?  Also improve the message to provide more details such as port number, or just reuse the error message retrieved from the log.

Reply:
Modified to reuse the error message retrieved from the log, it does reach the message after the stopServer(). 

However, occasionally it will still display `The server has stopped. Exiting dev mode` error from https://github.com/OpenLiberty/ci.common/blob/9cca388c94b432870873237b78138b5ff56846e0/src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java#L695

I will open a separate issue for this.

Reply:
Sure. Sounds like a timing issue.